reasoner/processor/AbstractProcessor.java

```diff
@@ -182,6 +185,36 @@ public Driver<Monitor> monitor() {
             return monitor;
         }
 
+        public ReasonerPerfCounters perfCounters() {
+            return perfCounters;
+        }
+
+        public static class ReasonerPerfCounters {
```

Comment:
Ok so after our discussion on usage of the Perf Counters, let's go back to your previous architecture: move this to `//reasoner/common/PerfCounters`. Then you can create the `PerfCounters` in the top-level `Reasoner` entry point, and pass it into the `ControllerRegistry`, which passes it through to the `Controller` and `Processor` contexts. Then we have wider access to it, such as being able to pass it into `Planner`.